internal/store/horizontalpodautoscaler.go

```diff
@@ -134,39 +134,41 @@ func hpaMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generat
 				for _, m := range a.Spec.Metrics {
 					var metricName string
 
-					var v [metricTargetTypeCount]int64
+					var v [metricTargetTypeCount]float64
 					var ok [metricTargetTypeCount]bool
 
 					switch m.Type {
 					case autoscaling.ObjectMetricSourceType:
 						metricName = m.Object.Metric.Name
 
-						v[value], ok[value] = m.Object.Target.Value.AsInt64()
+						if m.Object.Target.Value != nil {
+							v[value], ok[value] = float64(m.Object.Target.Value.MilliValue())/1000, true
+						}
 						if m.Object.Target.AverageValue != nil {
-							v[average], ok[average] = m.Object.Target.AverageValue.AsInt64()
+							v[average], ok[average] = float64(m.Object.Target.AverageValue.MilliValue())/1000, true
```

Comment:
Since we are now always storing `true` in `ok`, do we still need the `ok` variable? Can we try removing it to see if the tests break?

Reply:
Good idea! The `ok` variable seems useless. But removing it directly would break the unit test, since we set the length of the `v` to a constant, it would show the extra zeros (the default value).

Although I can judge whether the value of v[i] is zero (HPA target metric should be required to be greater than 0) to decide whether to display the corresponding metric, but I think this way is not very good.
So Could I change the type of `v` to the map type, and finally displayed the metric that exists in the map?

Reply:
I have push a new commit. How do you think?

Reply:
I think a map could work, but it might also create a flaky test. Iterating over a map does not guarantee that the order of the keys will always be the same. Can this lead to metrics being serialized in a different order each time a test is run? Could you run the tests with -count=100?

Reply:
Perfect!

/lgtm